Travis - fix shallow clone of XSMM 1.16

jeremylt opened the pull request:
This fixes my bad Travis clone for XSMM

jeremylt commented:
Verified - LIBXSMM builds fine now. Failures are due to Nek's bug https://github.com/Nek5000/Nek5000/pull/667

jedbrown commented:
If we pin `libxsmm`, we can drop the `_Bool` hack now: https://github.com/hfp/libxsmm/commit/143409cf00cf0a8c0e5e58676c0f56a80f4f4d2a

jeremylt commented:
Switched to LIBXSMM at the _Bool fix.

Also, I switched to before Nek5000's bug (until they review and merge my bugfix https://github.com/Nek5000/Nek5000/pull/667)
